mm/2s2h/Rando/Logic/Logic.h

```diff
@@ -98,6 +98,7 @@ extern std::unordered_map<RandoRegionId, RandoRegion> Regions;
 #define FOUND_ALL_FROGS                                                                  \
     (CHECK_WEEKEVENTREG(WEEKEVENTREG_33_01) && CHECK_WEEKEVENTREG(WEEKEVENTREG_32_40) && \
      CHECK_WEEKEVENTREG(WEEKEVENTREG_32_80) && CHECK_WEEKEVENTREG(WEEKEVENTREG_33_02))
+#define CAN_USE_ABILITY(ability) (Flags_GetRandoInf(ability - RI_ABILITY_SWIM + RANDO_INF_OBTAINED_SWIM))
```

Comment:
I think we should go this route, to be more in line with other macros such as `CAN_USE_MAGIC_ARROW`. This would allow us to use `CAN_USE_ABILITY(SWIM)` instead of listing the full RI every time and just generally look cleaner in the check tracker:

```suggestion
#define CAN_USE_ABILITY(ability) (Flags_GetRandoInf(RI_ABILITY_##ability - RI_ABILITY_SWIM + RANDO_INF_OBTAINED_SWIM))
```